src/code/padmgr.c

```diff
@@ -44,193 +56,349 @@ void PadMgr_UnlockReleaseControllerQueue(PadMgr* padmgr, OSMesgQueue* ctrlrqueue
+            LogUtils_LogThreadId(&D_80145B20, 0x17c); //"../padmgr.c"
+            osSyncPrintf(&D_80145B2C, padmgr->ctrlrIsConnected[i]); //"this->ctrlrIsConnected[i] = %d\n"
+            osSyncPrintf(&D_80145B48); //"\x1b[33m"
+            osSyncPrintf(&D_80145B50, temp_s1, &D_80145B64); //"padmgr: %dCon: %s\n" : "Overrun error occurred"
+            osSyncPrintf(&D_80145B80); //"\x1b[m"
+            break;
+        case 8:
+            input->current.input.button = 0;
+            input->current.input.x = 0;
+            input->current.input.y = 0;
+            temp_v0_2 = arg0 + phi_s2;
+            input->current.errno = padnow1->errno;
+            if (temp_v0_2->ctrlrIsConnected != 0){
+                temp_v0_2->ctrlrIsConnected = 0;
+                temp_v0_2->pakType = 0;
+                temp_v0_2->rumbleCounter = 0xff;
+                osSyncPrintf(&D_80145B84); //"\x1b[33m"
+                osSyncPrintf(&D_80145B8C, phi_s2 + 1, &D_80145BA0); //"padmgr: %dCon: %s\n" : "Do not respond"
+                osSyncPrintf(&D_80145BB0); //"\x1b[m"
+            }
+            break;
+        default:
+            LogUtils_LogThreadId(&D_80145BB4, 0x18c); //"../padmgr.c"
+            osSyncPrintf(&D_80145BC0, padnow1->errno); //"padnow1->errno = %x\n"
+            Fault_AddHungupAndCrash(&D_80145BD8, 0x18d); //"../padmgr.c"
+        }
+        input->pressed_diff.input.button = input->current.input.button & (input->prev.input.button ^ input->current.input.button);
+        input->released_adj.input.button = input->prev.input.button & (input->prev.input.button ^ input->current.input.button);
+        func_800FCC6C(input);
+        input->pressed_diff.input.x = (input->current.input.x - input->prev.input.x) + input->pressed_diff.input.x;
+        input->pressed_diff.input.y = (input->current.input.y - input->prev.input.y) + input->pressed_diff.input.y;
+    }
+    PadMgr_UnlockPadData(arg0);
 }
 #else
-#pragma GLOBAL_ASM("asm/non_matchings/code/padmgr/func_800C7948.s")
+#pragma GLOBAL_ASM("asm/non_matchings/code/padmgr/PadMgr_ProcessInputs.s")
 #endif
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/padmgr/func_800C7974.s")
+void PadMgr_HandleRetraceMsg(PadMgr* padmgr) {
+    OSMesgQueue* queue; // s1
+    u32 i;
+    u32 mask;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/padmgr/func_800C7C14.s")
+    queue = PadMgr_LockSerialMesgQueue(padmgr);
+    osContStartReadData(queue);
+    if (padmgr->retraceCallback) {
+        padmgr->retraceCallback(padmgr, padmgr->retraceCallbackValue);
+    }
+    osRecvMesg(queue, NULL, OS_MESG_BLOCK);
+    osContGetReadData(&padmgr->pads[0]);
+    if (padmgr->preNMIShutdown) {
+        bzero(&padmgr->pads[0], 0x18);
+    }
+    PadMgr_ProcessInputs(padmgr);
+    func_80103210(queue);
+    osRecvMesg(queue, NULL, OS_MESG_BLOCK);
+    func_8010328C(padmgr);
+    PadMgr_UnlockSerialMesgQueue(padmgr, queue);
+
+    mask = 0;                      // s2
+    for (i = 0; i - 4 != 0; ++i) { // s1
```

Comment:
I believe all these comments for registers aren't necessary anymore here since the function matches, so I suggest cleaning them up.
```suggestion
    mask = 0;
    for (i = 0; i - 4 != 0; ++i) {
```